### PR TITLE
Solve problem in xonsh activation script on Windows.

### PR DESCRIPTION
**Pull request by melund**
This fixes problem with hook preamble inserted into the xonsh activation script on Windows.

Since paths contain backslashes on windows we need to make the string raw. 

This following will not work due to the `\n` in the path
```
CONDA_EXE = "C:\Users\nemo\anaconda3\Scripts\conda.exe"
```
But this is better:
```
CONDA_EXE = r"C:\Users\nemo\anaconda3\Scripts\conda.exe"
```

The PR also removes the `native_path_to_unix`. Xonsh handles both path with both forward and backslashes, but it doesn't handle the CygWin versions of the Windows paths.

### CODE REVIEW

**Comment by msarahan** [comment]
It looks like the tests need to be updated for these raw strings.  I don't see anything wrong otherwise - just that the "correct" value specified by the tests needs to be updated to be a raw string.

You could also maybe just always code forward slashes, then use os.path.normpath to go back to Windows backslashes when necessary? https://pythonconquerstheuniverse.wordpress.com/2008/06/04/gotcha-%E2%80%94-backslashes-in-windows-filenames/

I'm ambivalent, but either way, please look into the failing tests.

**Comment by melund** [comment]
Yes. We really only want to use raw strings (or escape backslashes) for things that are paths. So maybe we could replace `path_conversion` with a function that converts backslashes to forward slashes on Win.

Not sure if `path_conversion` is called an every thing that is a path, but I will just try.

**Comment by scopatz** [comment]
@melund - I have fixed the merge conflicts here.  Can you please give me access to push to your fork? Thanks!

**Comment by melund** [comment]
@scopatz I have given you push rights.. I think ...

**Comment by scopatz** [comment]
OK! I have pushed up the changes (thanks!) but it seems like the tests fail.

**Comment by scopatz** [comment]
Yeah, it does seem like it

**Comment by scopatz** [comment]
Do you want to take another stab at fixing this up @melund?

File: conda/activate.py [hunk 1]
```diff
@@ -943,7 +954,7 @@ def __init__(self, arguments=None):
         super(XonshActivator, self).__init__(arguments)
 
     def _hook_preamble(self):
-        return '$CONDA_EXE = "%s"' % context.conda_exe
+        return '$CONDA_EXE = "%s"' % self.path_conversion(context.conda_exe)
```
Comment (melund) [inline conda/activate.py]: Ahh.  `self.path_conversion` goes to the parent class since `XonshActivator.path_conversion()` is a static method.

Comment (melund) [reply conda/activate.py]: Or maybe not. More likely just because the ``__init__`` overrides the ``path_conversion`` name.

**Comment by melund** [comment]
@scopatz. I think it is fixed now but the test seems to be failing in some completely unrelated places. I better try to pull from master and see if that fixes it.

**Comment by melund** [comment]
@soapy1 I think this is ready to be merged. The test failures are now unrelated. 

It is a followup to @scopatz's #9319 fix the other day.

**Comment by melund** [comment]
@scopatz Could you help nodge anyone to get this reviewed?

**Comment by msarahan** [comment]
Thanks @melund and @scopatz

**Pull request merged**